spec/javascripts/tests/GlobalNav_spec.js

```diff
@@ -0,0 +1,304 @@
+describe('GlobalNav', function() {
+  'use strict';
+
+  beforeEach(function(done) {
+    var self = this;
+
+    requirejs(
+        ['jquery', 'GlobalNav', 'mediaQueries', 'utilities', 'common'],
+        function($, GlobalNav, mediaQueries, utilities, common) {
+          self.$html = $(window.__html__['spec/javascripts/fixtures/GlobalNav.html']).appendTo('body');
+          self.component = self.$html.find('[data-dough-component="GlobalNav"]');
+          self.globalNav = GlobalNav;
+          self.mainContent = self.$html.find('#main');
+          self.mobileNavButton = (self.$html).find('[data-dough-mobile-nav-button]');
+          self.clumps = self.component.find('[data-dough-nav-clumps]');
+          self.subNavHeading = self.component.find('[data-dough-subnav-heading]');
+          self.mobileNavCloseButton = self.component.find('[data-dough-mobile-nav-close]');
+          self.mobileNavOverlay = self.component.find('[data-dough-mobile-nav-overlay]');
+          self.obj = new self.globalNav(self.component, self.component.data('dough-global-nav-config'));
+          self.delay = 250;
+
+          done();
+        }, done);
+  });
+
+  afterEach(function() {
+    this.$html.remove();
+  });
+
+  describe('Initialise component', function() {
+    it('calls the right methods when initialising', function() {
+      var setUpMobileInteractionStub = sinon.stub(this.obj, "_setUpMobileInteraction"),
+          setUpDesktopInteractionStub = sinon.stub(this.obj, "_setUpDesktopInteraction"),
+          setUpMobileAnimationStub = sinon.stub(this.obj, "_setUpMobileAnimation"),
+          setUpKeyboardEventsStub = sinon.stub(this.obj, "_setUpKeyboardEvents"),
+          bindEventsStub = sinon.stub(this.obj, "_bindEvents");
+
+      this.obj.init();
+
+      expect(setUpMobileInteractionStub.callCount).to.be.equal(1);
+      expect(setUpDesktopInteractionStub.callCount).to.be.equal(1);
+      expect(setUpMobileAnimationStub.callCount).to.be.equal(1);
+      expect(setUpKeyboardEventsStub.callCount).to.be.equal(1);
+      expect(bindEventsStub.callCount).to.be.equal(1);
+    });
+  });
+
+  describe('Bind Events', function() {
+    beforeEach(function() {
+      this.obj.init();
+    });
+
+    it('removes the uninialised class when component is loaded', function() {
+      expect(this.component.hasClass('uninitialised')).to.be.false;
+    });
+  });
+
+  describe('Mobile animation', function() {
+    beforeEach(function() {
+      this.obj.init();
+    });
+
+    it('toggles the no-transition class when viewport size changes', function() {
+      // This is not functional because I have been unable to simulate the resize event - DT
```

Comment:
@tomas-stefano I'm not sure about how to deal with this and ensure it gets done some time. 

@norm The problem with the resize event is that the component uses the media query class that is added to the <html> tag but this doesn't get read in the test scenario.